### ISSUE TITLE
The ID is not consistentcy

### ISSUE THREAD

**Issue by justcharliechan** [zo0r/react-native-push-notification#1923]
# Question
Why does the notification is different type in different function:

PushNotification.cancelLocalNotifications({ id: string });
PushNotification.localNotificationSchedule({ id: number});

**Comment by Dallas62** [comment]
Hi @justcharliechan 

It's `PushNotification.localNotificationSchedule({ id: string});`

>  id: 0, // (optional) Valid unique 32 bit integer specified **as string**. default: Autogenerated Unique ID

The type is cast if needed.

Regards

**Comment by justcharliechan** [comment]
> Hi @justcharliechan
> 
> It's `PushNotification.localNotificationSchedule({ id: string});`
> 
> > id: 0, // (optional) Valid unique 32 bit integer specified **as string**. default: Autogenerated Unique ID
> 
> The type is cast if needed.
> 
> Regards

Yes. We can force the notification id to string (ios + js), and use the hashCode in Java code.

**Comment by Dallas62** [comment]
Hi @justcharliechan 
This could be a good idea to test that, the only issue I see with this method is the collision.
A good hash method is required to limit the risk of collision.
Regards,

**Comment by theneekz** [comment]
@Dallas62 could you suggest an appropriate hash function in the readme?

**Comment by theneekz** [comment]
Perhaps a javascript implementation of java's hashCode:

```
/*eslint-disable no-bitwise*/
/*eslint-disable no-plusplus*/
const stringHash = (str) => {
  let hash = 0;
  let i = 0;
  const len = str.length;
  while (i < len) {
    hash = ((hash << 5) - hash + str.charCodeAt(i++)) << 0;
  }
  return hash;
};
export default stringHash;
```

Based off of [https://stackoverflow.com/a/33647870/13315676](https://stackoverflow.com/a/33647870/13315676) this converts a string into a signed 32 bit int for making ids for scheduled local notifications so that they can be cleared later.


Explanation:

-  "<<"" is the left shift bitwise operator [see MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Left_shift)

- "(hash << 5 - hash)" is a fast way of multiplying by 31

    - (x << 5) - x = (x * 32) - x = x * 31

    - why 31? [see quora answer](https://qr.ae/pGqNo0)

- "<< 0" is a fast way to convert a number in javascript to a signed 32 bit int [stackoverflow](https://stackoverflow.com/a/30673270/13315676)
